src/api/CommentApi.php

```diff
@@ -186,68 +194,72 @@ public function getActions(): array {
-		if ( !$user || $user->isAnon() ) {
-			return $this->dieWithError( [ 'comment-invaliduser' ] );
+	public function doAddToDefault(): void {
+		$userIdentity = $this->userIdentityLookup->getUserIdentityByUserId( $this->getMain()->getVal( 'user_id' ) );
+		if ( !$userIdentity || !$userIdentity->isRegistered() ) {
+			$this->dieWithError( [ 'comment-invaliduser' ] );
 		}
+
+		$user = $this->userFactory->newFromUserIdentity( $userIdentity );
 		$text = $this->getMain()->getVal( 'text' );
-		$inreply = $this->getInt( 'inReplyTo' );
+		$inReply = $this->getInt( 'inReplyTo' );
 
-		if ( MediaWikiServices::getInstance()->getUserOptionsLookup()->getIntOption( $user, 'comment-pref' ) ) {
+		if ( $this->userOptionsLookup->getIntOption( $user, 'comment-pref' ) ) {
 			$board = new CommentBoard( $user );
-			$commentSuccess = $board->addComment( $text, $this->getUser(), $inreply );
+			$commentSuccess = $board->addComment( $text, $this->getUser(), $inReply );
 			$this->getResult()->addValue( null, 'result', ( $commentSuccess ? 'success' : 'failure' ) );
-		} else {
-			// the recommended way of editing a local article was with WikiPage::doEditContent
-			// however there didn't seem to be an easy way to add a section rather than editing the entire content
-			$params = new DerivativeRequest(
-				$this->getRequest(),
-				[
-					'title' => 'User_talk:' . $user->getName(),
-					'action' => 'edit',
-					'section' => 'new',
-					'summary' => $this->getMain()->getVal( 'title' ),
-					'text' => $text,
-					'token' => $this->getMain()->getVal( 'token' ),
-				]
-			);
-			$api = new ApiMain( $params, true );
-			$api->execute();
-			// TODO: check the result object from the internal API call to determine success/failure status
-			$this->getResult()->addValue( null, 'result', 'success' );
+			return;
 		}
+
+		// the recommended way of editing a local article was with WikiPage::doEditContent
+		// however there didn't seem to be an easy way to add a section rather than editing the entire content
+		$params = new DerivativeRequest(
+			$this->getRequest(),
+			[
+				'title' => 'User_talk:' . $user->getName(),
+				'action' => 'edit',
+				'section' => 'new',
+				'summary' => $this->getMain()->getVal( 'title' ),
+				'text' => $text,
+				'token' => $this->getMain()->getVal( 'token' ),
+			]
+		);
+		$api = new ApiMain( $params, true );
+		$api->execute();
+		// TODO: check the result object from the internal API call to determine success/failure status
+		$this->getResult()->addValue( null, 'result', 'success' );
 	}
 
 	/**
 	 * Adds a new comment to a user's comment board on their Curse Profile page
 	 */
-	public function doAdd() {
-		$toUser = MediaWikiServices::getInstance()->getUserFactory()->newFromId( $this->getInt( 'user_id' ) );
-		if ( !$toUser || !$toUser->isAnon() ) {
-				$this->getResult()->addValue( null, 'result', 'failure' );
```

Comment:
IMO in original code `!$toUser->isAnon()` was a typo and there should be `$toUser->isAnon()`